pennylane/operation.py

```diff
@@ -906,6 +906,50 @@ def __matmul__(self, other):
 
         raise ValueError("Can only perform tensor products between observables.")
 
+    def obs_data(self):
+        r"""Extracts the data from an Observable/Tensor."""
+        obs = Tensor(self).non_identity_obs
+        tensor = set()
+
+        for ob in obs:
+            parameters = tuple(param.tostring() for param in ob.parameters)
+            tensor.add((ob.name, ob.wires, parameters))
+
+        return tensor
+
+    def compare(self, other):
+        r"""Compares two Observables/Tensors/qml.Hamiltonian objects to determine if they are equivalent.
+        """
+        val = False
+        if isinstance(other, (Observable, Tensor)):
+            val = self.obs_data() == other.obs_data()
+        if isinstance(other, qml.Hamiltonian):
+            val = other.obs_data() == self.obs_data()
+
+        return val
```

Comment:
```suggestion
        if isinstance(other, (Observable, Tensor)):
            return self.obs_data() == other.obs_data()
 
        if isinstance(other, qml.Hamiltonian):
            return other.obs_data() == self.obs_data()

        return False
```
Minutely more efficient, since if `other` is an observable/tensor, the second `isinstance()` is not called :)

Also, _should_ this method return `False` if `other` is not a known type, or raise an error? Raising an error makes slightly more sense to me

Reply:
Maybe a NotImplementedError would be best?